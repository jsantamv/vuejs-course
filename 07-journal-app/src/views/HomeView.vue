<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">

    <div>
      <h1>Hola Mundo</h1>
      <button @click="goToDayBook" class="btn btn-primary m-2">Primary</button>
      <button @click="goToDayBook" class="btn btn-secondary m-2">Secondary</button>
      <button @click="goToDayBook" class="btn btn-success m-2">Success</button>
    </div>
  </div>
</template>

<script>
  export default {
    methods: {
      goToDayBook() {
        this.$router.push({name: 'daybook'})
      }
    }
  }
</script>