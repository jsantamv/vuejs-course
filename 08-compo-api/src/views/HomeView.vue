<template>
  <div class="home">
    <h1>Reactive vs Ref</h1>

    <h3>{{ ironman }}</h3>
    <h3>{{ hulk }}</h3>

    <button @click="onChangeIronMan">Iron Man</button>
    <button @click="onChangeHulk">Hulk</button>


    <h3>Iroman esta vivo: {{ ironManIsAlive }}</h3>
    <h3>Hulk esta vivo: {{ hulkIsAlive }}</h3>

    <button @click="changeStatus">Cambiar Status</button>
  </div>
</template>

<script>
import { reactive, ref } from 'vue'
// @ is an alias to /src

export default {
  name: 'HomeView',

  setup() {
    const ironman = ref({ name: 'Tony', age: 50 })
    const hulk = reactive({ name: 'Bruce', age: 50 })

    let ironManIsAlive = reactive(false)
    const hulkIsAlive = ref(true)

    const changeStatus = () => {
      hulkIsAlive.value = !hulkIsAlive.value
      ironManIsAlive = !ironManIsAlive
    }

    return {
      ironman,
      hulk,
      ironManIsAlive,
      hulkIsAlive,

      changeStatus, 

      onChangeIronMan: () => {
        ironman.value.name = 'Tony Stark'
        ironman.value.age = 51
      },

      onChangeHulk: () => {
        hulk.name = 'Bruce Baner'
        hulk.age = 45
      }

    }
  }
}
</script>
