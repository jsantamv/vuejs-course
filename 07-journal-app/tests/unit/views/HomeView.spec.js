import { shallowMount } from '@vue/test-utils'
import HomeView from '@/views/HomeView'

describe('HomeView.vue', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(HomeView)
    })
    
    test('debe de renderizar el componente correctamente', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
