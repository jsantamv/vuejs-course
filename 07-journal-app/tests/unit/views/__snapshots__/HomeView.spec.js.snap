// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue debe de renderizar el componente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <div>
    <h1>Hola Mundo</h1><button class="btn btn-primary m-2">Primary</button><button class="btn btn-secondary m-2">Secondary</button><button class="btn btn-success m-2">Success</button>
  </div>
</div>
`;
